doc/holochain_101/src/zome/zome_functions.md

```diff
@@ -0,0 +1,154 @@
+
+A core feature of an HDK is that it will handle that native interface to Holochain, wrapping the underlying functions in easy to call, well defined and well documented functions that are native to the language that you're writing in!
+
+So within Zome code, by calling functions of the HDK, you can do powerful things with Holochain, like:
+  - Read and write data to and from the configured storage mechanism, and from the network
+  - Transport messages directly between nodes of a network
+  - Call functions in other Zomes, and even "bridged" DNA instances
+  - Use cryptographic functions like signing and verification to handle data security and integrity
+  - Emit "signals" containing data from a Zome, to a UI for example
+
+How these different functions work and how to use them will be covered throughout the rest of this chapter in detail. This article will provide a general overview of what Zome functions themselves are and how they work.
+
+Recall that Zomes will be written in diverse programming languages, any one that compiles to WebAssembly. Towards the bottom of this article, "Building in Rust" gives examples of what writing functions in Rust will be like. It is difficult to show what a function in WASM looks like, since even the "human-readable" version of WASM, WAT, is not highly readable.
+
+## DNA, Zomes, Capabilities, and Functions
+
+When Holochain loads a DNA file, to start an instance from it, it expects the presence of one or more Zomes in the definition. Here is a skeletal (incomplete) DNA JSON file that can illustrate this:
+
+```json
+{
+    "name": "test",
+    "zomes": {
+        "test_zome": {
+            "name": "test_zome",
+            "capabilities": {
+                "test_capability": {
+                    "type": "public",
+                    "fn_declarations": []
+                }
+            },
+            "code": {
+                "code": "AAECAw=="
+            }
+        }
+    }
+}
+```
+
+This theoretical DNA has one Zome, "test_zome". However, it has no functions. Note that the nested `fn_declarations` property is an empty array.
+
+There are few things to learn from this DNA JSON. The first, that is simple to explain, is that the code, Base64 encoded WASM, is actually embedded in the Zome's definition, nested under `code.code`. All the functions Holochain expects to be implemented need to be encapsulated within that WASM code.
+
+The second is that even outside of the WASM code, Holochain expects a certain level of visibility into the functions contained within, at least the ones meant to be called via Holochain (as oppose to private/internal functions).
+
+There are at least two reasons for this:
+- to define a permission based system for those functions
+- to be able to reason about data inputs and outputs for those functions
+
+These will both be discussed below.
+
+## Capabilities
```

Comment:
for 0.0.4 I will be implementing capabilities for real, which includes separating out the function declarations from the capabilities, which simply maps a capability relationship onto a set of functions.  For the structure of the MD book I think we should split capabilities into it's own file, which is about the security model (and the relations to bridging).   I'm happy to do this in one of the capabilities PRs as I implement this, but it will be useful get this PR into develop so I can work from what you've got.

Reply:
Sounds great. I originally had Capabilities mapped out in a separate article, but because of its minimal implementation atm I changed that. Would be happy to see it changed back when its fleshing out